roms_tools/plot.py

```diff
@@ -597,13 +631,464 @@ def _add_field_to_ax(
+        Useful to exclude ghost points. Default is True.
+
+    depth_contours : bool, optional
+        If True, add contours of constant bathymetry to horizontal plots. Ignored otherwise.
+        Default is False.
+
+    layer_contours : bool, optional
+        If True, show vertical layer boundaries in vertical sections. Default is False.
+
+    max_nr_layer_contours : int, optional
+        Maximum number of vertical layer contours to draw. Default is 10.
+
+    use_coarse_grid : bool, optional
+        Use precomputed coarse-resolution grid. Default is False.
+
+    with_dim_names : bool, optional
+        Add grid dimension names (`xi`, `eta`) to the outer plot edges. Only for 2D plots. Default is False.
+
+    ax : Axes, optional
+        Matplotlib axes object. If None, a new figure is created. Default is None.
+
+    save_path : str, optional
+        File path to save the plot. If None, the plot is shown interactively. Default is None.
+
+    cmap_name : str, optional
+        Name of matplotlib colormap. Default is "YlOrRd".
+
+    add_colorbar : bool, optional
+        Whether to add a colorbar to the plot. Default is True.
+
+    Returns
+    -------
+    None
+        The function generates and shows or saves a plot. No value is returned.
+
+    Raises
+    ------
+    ValueError
+        - If no plotting dimensions are specified for 3D fields.
+        - If conflicting dimensions are specified (e.g., both `s` and `depth`).
+        - If incompatible combinations of grid indices and geographic coordinates are provided.
+        - If specified `eta` or `xi` indices are out of bounds.
+        - If boundary indices are used when `include_boundary=False`.
+    """
+
+    # Input checks
+    _validate_plot_inputs(field, s, eta, xi, depth, lat, lon, include_boundary)
+
+    if "straddle" not in grid_ds.attrs:
+        raise AttributeError("Grid dataset must have a 'straddle' attribute.")
+    straddle = grid_ds.attrs["straddle"]
+
+    # Get horizontal dimensions and grid location
+    horizontal_dims_dict = {
+        "rho": {"eta": "eta_rho", "xi": "xi_rho"},
+        "u": {"eta": "eta_rho", "xi": "xi_u"},
+        "v": {"eta": "eta_v", "xi": "xi_rho"},
+    }
+    for loc, horizontal_dims in horizontal_dims_dict.items():
+        if all(dim in field.dims for dim in horizontal_dims.values()):
+            break
+
+    # Convert relative to absolute indices
+    def _get_absolute_index(idx, field, dim_name):
+        index = field[dim_name].isel(**{dim_name: idx}).item()
+        return index
+
+    if eta is not None and eta < 0:
+        eta = _get_absolute_index(eta, field, horizontal_dims["eta"])
+    if xi is not None and xi < 0:
+        xi = _get_absolute_index(xi, field, horizontal_dims["xi"])
+    if s is not None and s < 0:
+        s = _get_absolute_index(s, field, "s_rho")
+
+    # Set spatial coordinates
+    if use_coarse_grid:
```

Comment:
Consider a function `get_spatial_coords` that can be tested individually

Reply:
Agreed. Let's leave it for a future PR?